tests/pyfunc/test_pyfunc_schema_enforcement.py

```diff
@@ -2635,3 +2652,167 @@ def predict(self, context, model_input, params=None):
     result = json.loads(response.content.decode("utf-8"))["predictions"]
     expected_result = df.to_dict(orient="records")
     np.testing.assert_equal(result, expected_result)
+
+
+def test_enforce_schema_spark_dataframe():
+    spark = SparkSession.builder.appName("test").getOrCreate()
```

Comment:
```suggestion
@pytest.fixture(scope="module")
def spark():
    with SparkSession.builder.getOrCreate() as session:
        yield session


def test_enforce_schema_spark_dataframe(spark):
```

We can define a module-scoped fixture to create a spark session so we don't need to create it in each test.